src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -190,115 +133,52 @@ public function invalidateTags(array $tags): bool
-                if (($knownTagVersions[$itemTag][0] ?? 0.0) < $now || $knownTagVersions[$itemTag][1] !== $itemTagVersion) {
-                    break 2;
-                }
-            }
-
+        foreach ($this->getItems([$key]) as $item) {
             return $item;
         }
-        $knownTagVersions = null;
-
-        $tagVersions = $this->getTagVersions(array_keys($itemTagVersions));
-        foreach ($itemTagVersions as $itemTag => $itemTagVersion) {
-            if (!isset($tagVersions[$itemTag]) || $tagVersions[$itemTag] !== $itemTagVersion) {
-                (self::$unsetCacheItem)($item);
-                break;
-            }
-        }
-
-        return $item;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getItems(array $keys = []): iterable
     {
-        $items = $itemIdsMap = $itemTagVersions = $tagVersions = [];
         $commit = false;
 
         foreach ($keys as $key) {
-            $itemIdsMap[$this->getPrefixedKey($key)] = $key;
-            $commit = $commit || isset($this->deferred[$key]);
+            if ($commit = isset($this->deferred[$key])) {
+                break;
+            }
         }
 
         if ($commit) {
             $this->commit();
         }
 
-        $validateAgainstKnownTagVersions = !empty($this->knownTagVersions);
-        $f = self::$unpackCacheItem;
-        foreach ($this->pool->getItems(array_keys($itemIdsMap)) as $itemId => $item) {
-            $key = $itemIdsMap[$itemId];
-            $itemTagVersions[$key] = $t = ($f)($item, $key);
-            $items[$key] = $item;
-            if (!$t) {
-                continue;
-            }
-            $tagVersions += $t;
-            if ($validateAgainstKnownTagVersions) {
-                foreach ($t as $tag => $tagVersion) {
-                    if ($tagVersions[$tag] !== $tagVersion) {
-                        $validateAgainstKnownTagVersions = false;
-                        break;
-                    }
-                }
-            }
+        try {
+            $items = $this->pool->getItems($keys);
+        } catch (InvalidArgumentException $e) {
+            $this->pool->getItems($keys); // Should throw an exception
+
+            throw $e;
         }
-        $itemIdsMap = null;
 
-        return $this->generateItems($items, $itemTagVersions, $tagVersions, $validateAgainstKnownTagVersions);
+        return $this->generateItems($items);
     }
 
     /**
      * {@inheritdoc}
      */
     public function clear(string $prefix = ''): bool
     {
-        if ($this->pool instanceof AdapterInterface) {
-            $isPoolCleared = $this->pool->clear(self::ITEM_PREFIX.$prefix);
-        } else {
-            $isPoolCleared = $this->pool->clear();
-        }
-
-        if ($this->tags instanceof AdapterInterface) {
-            $isTagPoolCleared = $this->tags->clear(static::TAG_PREFIX.$prefix);
-        } else {
-            $isTagPoolCleared = $this->tags->clear();
-        }
```

Comment:
Second read: not clearing tags was on purpose. Tags could be shared among several item pools and clearing them for one pool would break this use case. If one wants to clear tags, they should call the method on the tags pool directly.